editor/store/reducer.js

```diff
@@ -656,17 +656,12 @@ export function preferences( state = PREFERENCES_DEFAULTS, action ) {
 					id += '/' + block.attributes.ref;
 				}
 
-				const isSameAsInsert = ( { name, ref } ) => name === insert.name && ref === insert.ref;
-
 				return {
 					...prevState,
-					recentInserts: [
-						insert,
-						...reject( prevState.recentInserts, isSameAsInsert ),
-					],
 					insertUsage: {
 						...prevState.insertUsage,
 						[ id ]: {
+							time: Date.now(),
```

Comment:
Pedantry: `Date.now` is impure:

https://en.wikipedia.org/wiki/Functional_programming#Pure_functions

>If a pure function is called with arguments that cause no side-effects, the result is constant with respect to that argument list ... i.e., if calling the pure function again with the same arguments returns the same result.

Violating a core principle of Redux:

>Reducers are just pure functions that take the previous state and an action, and return the next state.

https://redux.js.org/introduction/three-principles#changes-are-made-with-pure-functions

Reply:
You're totally right. I'll move this to the action creator.